Update RG description in release checklist (#3741)
GenericMappingTools/pygmt: .github/ISSUE_TEMPLATE/4-release_checklist.md

- [ ] Bump PyGMT version on https://github.com/GenericMappingTools/try-gmt (after conda-forge update)
- [ ] Announce the release on:
  - [ ] GMT [forum](https://forum.generic-mapping-tools.org/c/news/) (do this announcement first! Requires moderator status)

>   - [ ] [ResearchGate](https://www.researchgate.net) (after forum announcement, add new version as research item (previously downloaded as zip-file from the [Releases](https://github.com/GenericMappingTools/pygmt/releases) section) via the **code** category, be sure to include the corresponding new Zenodo DOI)

Reviewer: I think it makes sense to mention this here.
Maybe this formulation without double brackets is easier to read, but also OK with another formulation.

```suggestion
  - [ ] [ResearchGate](https://www.researchgate.net) (after forum announcement; download the ZIP file of the new release from the [GitHub Releases section](https://github.com/GenericMappingTools/pygmt/releases) and add it as research item via the **code** category, be sure to include the corresponding new Zenodo DOI)
```
